data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/coordination/state/StreamProgressState.java

```diff
@@ -16,6 +16,12 @@ public class StreamProgressState {
     @JsonProperty("currentPosition")
     private BinlogCoordinate currentPosition;
 
+    @JsonProperty("currentLsn")
```

Comment:
Consider creating new class to separate MySQL vs Postgres properties.

Reply:
This model is used in serialization/deserialization by the source coordinator, it's a bit more involved to separate into two classes. I added a TODO in code and will address it in followup PRs.